modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/runtime/compatibility/DefaultForwardCompatibilityHelper.java

```diff
@@ -6,7 +6,41 @@
  */
 package org.mule.runtime.module.extension.api.runtime.compatibility;
 
-import org.mule.sdk.api.utils.ForwardCompatibilityHelper;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.internal.profiling.InternalProfilingService;
+import org.mule.runtime.extension.api.runtime.parameter.CorrelationInfo;
+import org.mule.runtime.module.extension.internal.runtime.parameter.ImmutableCorrelationInfo;
+import org.mule.runtime.module.extension.internal.runtime.resolver.DistributedTraceContextManagerResolver;
+import org.mule.runtime.module.extension.internal.runtime.source.trace.DefaultDistributedSourceTraceContext;
+import org.mule.sdk.api.runtime.source.DistributedTraceContextManager;
+import org.mule.sdk.compatibility.api.utils.ForwardCompatibilityHelper;
+
+import javax.inject.Inject;
 
 public class DefaultForwardCompatibilityHelper implements ForwardCompatibilityHelper {
+
+  @Inject
+  private InternalProfilingService profilingService;
+
+  public DistributedTraceContextManager getDistributedTraceContextManager(CorrelationInfo correlationInfo) {
+    if (correlationInfo instanceof ImmutableCorrelationInfo) {
+      CoreEvent event = ((ImmutableCorrelationInfo) correlationInfo).getEvent();
+      DistributedTraceContextManagerResolver argumentResolver =
+          new DistributedTraceContextManagerResolver(profilingService.getCoreEventTracer());
+      return argumentResolver.resolve(event);
+    } else {
+      return new DefaultDistributedSourceTraceContext();
+    }
+  }
+
+  public DistributedTraceContextManager getDistributedTraceContextManager(org.mule.sdk.api.runtime.parameter.CorrelationInfo correlationInfo) {
+    if (correlationInfo instanceof ImmutableCorrelationInfo) {
+      CoreEvent event = ((ImmutableCorrelationInfo) correlationInfo).getEvent();
```

Comment:
As this is implemented, the only way for the resolvers that have something injected from mule context container is to pass it through the correlationInfo. Not sure if this is the expected approach.